examples/nodejs/lambda-examples/cloudwatch-metrics/infrastructure/lib/momento-metrics-stack.ts

```diff
@@ -0,0 +1,310 @@
+          image: ecs.ContainerImage.fromDockerImageAsset(imageAsset),
+          logging: new ecs.AwsLogDriver({
+            logGroup: logGroup,
+            streamPrefix: "MomentoMetricsMiddlewareCDKExample"
+          })
+        });
+        taskDefinition.addToTaskRolePolicy(new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            "secretsmanager:ListSecrets",
+            "secretsmanager:GetSecretValue",
+            "logs:PutLogEvents",
+            "logs:CreateLogStream",
+            "ecs:*"
+          ],
+          resources: ["*"],
+        }));
+        taskDefinition.addToExecutionRolePolicy(new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            "secretsmanager:ListSecrets",
+            "secretsmanager:GetSecretValue",
+            "logs:PutLogEvents",
+            "logs:CreateLogStream",
+            "ecs:*"
+          ],
+          resources: ["*"],
+        }));
+
+        new ecs.FargateService(this, 'MomentoMetricsECSFargateService', {
+          cluster,
+          taskDefinition,
+        });
+        break;
+      }
+      case exampleApp.Custom: {
+        console.log('Skipping to dashboard creation');
+        break;
+      }
+      default: {
+        throw new Error('Unimplemented CDK stack application');
+      }
+    }
+
+    logGroup.addMetricFilter('ExampleMetricFilterDuration', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'Duration (Latency)',
+      filterPattern: FilterPattern.literal('{$.duration > 0}'),
+      metricValue: '$.duration',
+      unit: Unit.MILLISECONDS,
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterRequestSize', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'Request Size (bytes)',
+      filterPattern: FilterPattern.literal('{$.requestSize >= 0}'),
+      metricValue: '$.requestSize',
+      unit: Unit.BYTES,
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterResponseSize', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'Response Size (bytes)',
+      filterPattern: FilterPattern.literal('{$.responseSize >= 0}'),
+      metricValue: '$.responseSize',
+      unit: Unit.BYTES,
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterGrpcStatusCode', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'GRPC Status Code',
+      filterPattern: FilterPattern.literal('{ $.status >= 0 }'),
+      metricValue: '$.status',
+    });
+
+    const dashboard = new Dashboard(this, 'MomentoMetricsCDKExampleDashboard', {
+      dashboardName: 'MomentoMetricsCDKExampleDashboard',
+      defaultInterval: cdk.Duration.hours(1),
+    });
+
+    const latency = new GraphWidget({
+      left: [
+        new Metric({
+          metricName: 'Duration (Latency)',
+          namespace: 'MomentoMetricsCDKExample',
+          region: 'us-west-2',
```

Comment:
the `cdk.Stack` object should have a variable that you can read to get the region string. I can't remember if it's `this.region` or `this.env.region` or what, but it should be available.